builtin/credential/ldap/backend.go

```diff
@@ -212,70 +251,89 @@ func getUserDN(cfg *ConfigEntry, c *ldap.Conn, bindDN string) (string, error) {
 	return userDN, nil
 }
 
+/*
+ * getLdapGroups queries LDAP and returns a slice describing the set of groups the authenticated user is a member of.
+ *
+ * The search query is constructed according to cfg.GroupFilter, and run in context of cfg.GroupDN.
+ * Groups will be resolved from the query results by following the attribute defined in cfg.GroupAttr.
+ *
+ * cfg.GroupFilter is a go template and is compiled with the following context: [UserDN, Username]
+ *    UserDN - The DN of the authenticated user
+ *    UserName - The Username of the authenticated user
```

Comment:
This is sometimes `UserName` and sometimes `Username` -- please pick one and ensure consistency :-)